CMakeLists.txt

```diff
@@ -102,6 +102,14 @@ if(PCAPPP_ENABLE_PCAP_SET_DIRECTION)
   add_definitions(-DHAS_SET_DIRECTION_ENABLED)
 endif()
 
+option(PCAPPP_ENABLE_CLANG-TIDY "Run Clang-Tidy static analysis during build" OFF)
+
+if (PCAPPP_ENABLE_CLANG-TIDY)
+  find_program(CLANG_TIDY_EXE NAMES "clang-tidy" REQUIRED)
+  set(CLANG_TIDY_COMMAND "${CLANG_TIDY_EXE}" "--fix" "--checks=modernize-use-nullptr")
```

Comment:
- do we only want to check `nullptr`? Maybe enable all checks?
- Do we always want the `--fix` flag or should it be configurable?

Reply:
For the moment we only check for nullptr yes, I would like toenable the other checks progressively in another MR

Reply:
ok sounds reasonable, what about the `--fix` flag?